hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -168,7 +168,7 @@ public class HoodieTableMetaClient implements Serializable {
   private FileSystemRetryConfig fileSystemRetryConfig = FileSystemRetryConfig.newBuilder().build();
   protected HoodieMetaserverConfig metaserverConfig;
   private HoodieTimeGeneratorConfig timeGeneratorConfig;
-  private Option<HoodieIndexMetadata> indexMetadataOpt = Option.empty();
+  private Option<HoodieIndexMetadata> indexMetadataOpt;
```

Comment:
Since index definition cab be mutable, should we also expose a `reloadIndexDefinition` method?